### PR TITLE
feat(*): Adds custom time package for better marshalling

### LINKED ISSUE #6163: helm3: deleted in release info has strange default
I just did a `helm install wordpress stable/wordpress` with helm3-alpha2.
Then I looked at the release secret `k get secret wordpress.v1 -o jsonpath="{.data.release}" | base64 -D | base64 -D | gzip -d | jq`

`
{
  "name": "wordpress",
  "info": {
    "first_deployed": "2019-08-05T13:52:08.317608+02:00",
    "last_deployed": "2019-08-05T13:52:08.317608+02:00",
    "deleted": "0001-01-01T00:00:00Z",
    "Description": "Install complete",
    "status": "deployed", ...
`

As you can see the deleted timestamp has some strange default value. I think it should be an empty string or something like that.

Best regards,
Tobi

### PR DESCRIPTION
**Pull request by thomastaylor312**
This package mainly exists to workaround an issue in Go where the serializer doesn't omit an empty value for time: https://github.com/golang/go/issues/11939. This replaces all release and hook object time references with the new time package so things actually marshal correctly.

I also have a second commit that wraps most of the stdlib `Time` object methods so we can use our time wrapper without any weird conventions. This can be popped off and discarded if not desired.

I fully admit this adds some code and mental overhead, but in service of good output for users of the CLI and SDK. If we consider this trade off too much, I have no problem just closing this PR. See the thread starting [here](https://github.com/helm/helm/pull/6229#issuecomment-541936523) for additional context.

Supersedes: #6229 
Fixes #6163

### CODE REVIEW

File: cmd/helm/history.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ package main
 import (
 	"fmt"
 	"io"
-	"time"
+	stdtime "time"
```
Comment (bacongobbler) [inline cmd/helm/history.go]: Would it make sense to call helm's time package `helmtime` and keep `time` as `time`? I feel that may be less confusing to the reader that `stdtime` isn't some other third party package. Also, it opens up the opportunity for us to say

> it's `helmtime`

:trollface:

**Comment by bacongobbler** [comment]
ewww, but it works so :man_shrugging: 

What if we made `pkg/time` internal? `internal/time`

File: pkg/release/info.go [hunk 2]
```diff
@@ -24,9 +26,9 @@ type Info struct {
 	// LastDeployed is when the release was last deployed.
 	LastDeployed time.Time `json:"last_deployed,omitempty"`
 	// Deleted tracks when this object was deleted.
-	Deleted time.Time `json:"deleted,omitempty"`
+	Deleted time.Time `json:"deleted"`
 	// Description is human-friendly "log entry" about this release.
-	Description string `json:"Description,omitempty"`
+	Description string `json:"description,omitempty"`
```
Comment (bacongobbler) [inline pkg/release/info.go]: :+1:

**Comment by thomastaylor312** [comment]
@bacongobbler I was going to do the same thing, but because this time type is used in external packages, it needs to be exported so consumers of the SDK can use it. However, I am in favor of adding a very clear warning to the documentation for the package that it is not subject to the same backwards compatibility guarantees. What do you think?

File: cmd/helm/history.go (hunk 1 above)
Comment (thomastaylor312) [reply cmd/helm/history.go]: I thought about that. Since you brought it up, I'll change it

**Review by hickeyma — COMMENTED** [review]
Thanks for this @thomastaylor312, the changes look good.

I am however, not so sure the gains of adding wrapper code like this based on a PR pending in the other package. I am just not sure do we want to inherit this. It would be worth getting some feedback from the other maintainers also.

**Comment by thomastaylor312** [comment]
@hickeyma There isn't a PR pending for this issue in Go. The issue has been open for 4 years with multiple closed PRs attached to it, and the proposal seems to be stuck based on the labels (`Proposal-Hold`), so that isn't my worry.

My main worry is around having this code to maintain with the trade off in value in provides. I _really_ don't like marshaling out random timestamp strings that represent a zero value in Go, but are nonsense to anyone else. We could add custom marshaling to each object with a timestamp, but I think that would be even more brittle than this option (it was what I did on my first try). 

tl;dr: Is the user experience here (sane empty values) worth the code overhead?

**Review by hickeyma — APPROVED** [review]
I appreciate your feedback on my reservation.  It seems like good value to add then. LGTM, thanks @thomastaylor312

File: cmd/helm/history.go (hunk 1 above)
Comment (thomastaylor312) [reply cmd/helm/history.go]: Done

**Review by bacongobbler — APPROVED** [review]
